app/src/main/java/org/simple/clinic/home/HomeScreenEffectHandler.kt

```diff
@@ -24,8 +29,21 @@ class HomeScreenEffectHandler @AssistedInject constructor(
       .subtypeEffectHandler<HomeScreenEffect, HomeScreenEvent>()
       .addAction(OpenFacilitySelection::class.java, uiActions::openFacilitySelection, schedulersProvider.ui())
       .addTransformer(LoadCurrentFacility::class.java, loadCurrentFacility())
+      .addTransformer(LoadOverdueAppointmentCount::class.java, loadOverdueAppointmentCount())
       .build()
 
+  private fun loadOverdueAppointmentCount(): ObservableTransformer<LoadOverdueAppointmentCount, HomeScreenEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .flatMap { (facility) ->
```

Comment:
We should ideally be using `.switchMap` by default and only using `.flatMap` where it is required.

Reply:
Correct me if I am wrong, we need to use `switchMap` to get the latest observable stream right? Does that mean `flatMap` will retain older streams?

Reply:
Yes, that's exactly what `flatMap` does.

Reply:
If you look at the source code for [`ObservableFlatMap`](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/observable/ObservableFlatMap.java) and [`ObservableSwitchMap`](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/observable/ObservableSwitchMap.java), you'll see the different between the two.